generic/ubuntu1804 default locale is Latin-1

I've discovered that `/etc/default/locale` on generic/ubuntu1804 contains

    LANG="en_US"
    LANGUAGES="en_US:"

This causes problems, e.g. I cannot create PostgreSQL databases using UTF-8 because the system locale uses Latin-1.

(Some of the problems are masked when you use `vagrant ssh` because SSH copies LANG and LC_* from your host system.)

I believe stock Ubuntu defaults to UTF-8 locales and I think the generic boxes should too.

The workaround I applied in the mean time was rewriting /etc/default/locale in my Vagrant provisioning scrpts to contain `LANG="en_US.UTF-8"`.  For Ubuntu's PostgreSQL specifically this needs to be done before apt installing the postgresql-server package (because that package's install script creates a cluster configuration that remembers the system locale used during creation, for I don't know what reason -- I suppose the on-disk data structures assume a particular collation order or something).

I'm not entirely sure I understand what you mean by `COLLATION` -- the strings passed to `debian-installer/locale` are glibc locale names.  The format for them (documented in [the setlocale manual page](http://manpages.ubuntu.com/manpages/disco/en/man3/setlocale.3.html#description)) is `language[_TERRITORY][.charset][@modifier]` with the territory, charset, and modifier parts being optional.  You can run `locale -a` to see a list of locale names supported by the system.  (The charset part is passed through a normalization step, so that both UTF-8 and utf8 mean the same thing.  Modifiers were used by things like adding the Euro symbol into an otherwise Latin-1 locale in the bad old days when 8-bit charsets reigned supreme.)

So, I'm reasonably certain that changing `en_US` to `en_US.UTF-8` would work for all ubuntu and debian boxes, but I agree that it would be irresponsible to push new images without testing.  I'd appreciate some help with that: I read the README, ran `res/providers/packer.sh`, and it failed for me:
```
# github.com/hashicorp/packer/common/net
../../hashicorp/packer/common/net/configure_port.go:61:2: undefined: net.ListenConfig
error: pathspec 'except_post_processor_tests' did not match any file(s) known to git
res/providers/packer.sh: line 50: scripts/build.sh: No such file or directory
```

The failure confuses me because ~/go/src/github.com/hashicorp/packer/scripts/build.sh most definitely exists.  It fails, though:

```
$ cd ~/go/src/github.com/hashicorp/packer 
$ export GOPATH=$HOME/go/
$ PATH=$GOPATH/bin:$PATH
$ XC_ARCH=amd64 XC_OS="windows darwin linux" scripts/build.sh
3 errors occurred:
--> linux/amd64 error: exit status 2
Stderr: # github.com/hashicorp/packer/common/net
common/net/configure_port.go:61:2: undefined: net.ListenConfig

--> darwin/amd64 error: exit status 2
Stderr: # github.com/hashicorp/packer/common/net
common/net/configure_port.go:61:2: undefined: net.ListenConfig

--> windows/amd64 error: exit status 2
Stderr: # github.com/hashicorp/packer/common/net
common/net/configure_port.go:61:2: undefined: net.ListenConfig

==> Copying binaries for this platform...
find: ‘./pkg/linux_amd64’: Toks failas ar aplankas neegzistuoja

==> Results:
viso 0
```

--- a/http/generic.debian10.vagrant.cfg
+++ b/http/generic.debian10.vagrant.cfg
@@ -2,7 +2,7 @@ choose-mirror-bin mirror/http/proxy string
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.debian8.vagrant.cfg
+++ b/http/generic.debian8.vagrant.cfg
@@ -7,7 +7,7 @@ d-i mirror/http/directory string /debian
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0 
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.debian9.vagrant.cfg
+++ b/http/generic.debian9.vagrant.cfg
@@ -2,7 +2,7 @@ choose-mirror-bin mirror/http/proxy string
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0 
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.ubuntu1604.vagrant.cfg
+++ b/http/generic.ubuntu1604.vagrant.cfg
@@ -1,7 +1,7 @@
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0 
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.ubuntu1610.vagrant.cfg
+++ b/http/generic.ubuntu1610.vagrant.cfg
@@ -1,7 +1,7 @@
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0 
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.ubuntu1704.vagrant.cfg
+++ b/http/generic.ubuntu1704.vagrant.cfg
@@ -1,7 +1,7 @@
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0 
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.ubuntu1710.vagrant.cfg
+++ b/http/generic.ubuntu1710.vagrant.cfg
@@ -1,7 +1,7 @@
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.ubuntu1804.vagrant.cfg
+++ b/http/generic.ubuntu1804.vagrant.cfg
@@ -1,7 +1,7 @@
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.ubuntu1810.vagrant.cfg
+++ b/http/generic.ubuntu1810.vagrant.cfg
@@ -1,7 +1,7 @@
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

--- a/http/generic.ubuntu1904.vagrant.cfg
+++ b/http/generic.ubuntu1904.vagrant.cfg
@@ -1,7 +1,7 @@
 d-i base-installer/kernel/override-image string linux-server
 d-i keyboard-configuration/xkb-keymap select us
 d-i time/zone string US/Pacific
-d-i debian-installer/locale string en_US
+d-i debian-installer/locale string en_US.UTF-8
 d-i debian-installer/add-kernel-opts string net.ifnames=0 biosdevname=0
 d-i finish-install/reboot_in_progress note
 d-i grub-installer/bootdev string default

